packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -131,26 +136,55 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         _electionStore().nextNominationPeriodPercent = percent;
     }
 
-    function elect(address[] memory candidates) external view override {
-        uint seatCount = _electionStore().nextSeatCount;
+    function elect(address[] memory candidates) external override {
+        ElectionStore storage store = _electionStore();
+        ElectionData storage electionData = store.electionData;
 
-        if (candidates.length != seatCount) {
+        if (candidates.length == 0) {
             revert InvalidCandidatesCount();
         }
 
-        // TODO: if msg.sender already voted, rollback previous votes.
+        if (electionData.addressVoted[msg.sender]) {
+            // TODO: if msg.sender already voted, rollback previous votes and allow to re-vote
+            revert AlreadyVoted();
+        }
 
-        // TODO: Assign votes to each address (same to all)
-        // uint votePower = ERC20(_electionStore().electionTokenAddress).balanceOf(msg.sender);
+        // The voting power is the amount of votes we are going to assign to the given candidates
+        uint votePower = ERC20(store.electionTokenAddress).balanceOf(msg.sender);
+
+        for (uint i = 0; i < candidates.length; i++) {
+            address candidate = candidates[i];
+
+            // Check that the candidate is a nominee
+            if (electionData.nomineeIndexes[candidate] == 0) {
+                revert InvalidCandidate(candidate);
+            }
+
+            // Check that all the values on the candidates Array are unique
+            if (i < candidates.length - 1) {
```

Comment:
Loops are bad in Solidity. Nested loops are really bad!

I think that this check can be performed outside of this loop, probably with a utility function that sweeps the array and reverts if it counts more than one occurrence for any given entry. I.e. it doesn't need to know which two indices collide, just that they do.